user/super/com/google/gwt/emul/java/lang/Integer.java

```diff
@@ -184,24 +184,17 @@ public static int reverseBytes(int i) {
   }
 
   public static int rotateLeft(int i, int distance) {
-    while (distance-- > 0) {
-      i = i << 1 | ((i < 0) ? 1 : 0);
-    }
-    return i;
+    int rotateValue = distance % SIZE;
```

Comment:
I feel like you should try shifting by any value outside the range this guarantees (which isn't 0 to 31, by the way, it's -31 to 31) in the browser's JS Console. You shouldn't just take my word for it, but the modulus is totally unnecessary here; shifts are implicitly done with a mask of 31 in JS, and in Java for ints.

I would not be at all surprised if performing modulus on JS Number types is slower than the loop before, and it's completely unnecessary. A mask would arguably be more clear, and wouldn't permit negative `rotateValue`s if that's a concern. A mask isn't needed either.